src/app/pages/rpc-sign-stacks-message/use-rpc-sign-stacks-message.ts

```diff
@@ -0,0 +1,158 @@
+import { useMemo } from 'react';
+import { useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { RpcErrorCode } from '@btckit/types';
+import { StacksNetwork } from '@stacks/network';
+import { deserializeCV } from '@stacks/transactions';
+
+import { logger } from '@shared/logger';
+import { makeRpcErrorResponse, makeRpcSuccessResponse } from '@shared/rpc/rpc-methods';
+import {
+  UnsignedMessage,
+  isSignableMessageType,
+  isStructuredMessageType,
+  isUtf8MessageType,
+} from '@shared/signature/signature-types';
+import { closeWindow } from '@shared/utils';
+
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { useDefaultRequestParams } from '@app/common/hooks/use-default-request-search-params';
+import { useRejectIfLedgerWallet } from '@app/common/rpc-helpers';
+import { useWalletType } from '@app/common/use-wallet-type';
+import { createDelay } from '@app/common/utils';
+import {
+  StructuredPayload,
+  Utf8Payload,
+} from '@app/features/stacks-message-signer/stacks-message-signing';
+import { useMessageSignerStacksSoftwareWallet } from '@app/features/stacks-message-signer/stacks-message-signing.utils';
+
+// TODO: We sure we want users to wait for 1 second for nothing?
+const improveUxWithShortDelayAsSigningIsSoFast = createDelay(1000);
+
+const getNetwork = (networkName: string | null) => {
+  if (
+    networkName === 'mainnet' ||
+    networkName === 'testnet' ||
+    networkName === 'devnet' ||
+    networkName === 'mocknet'
+  ) {
+    return StacksNetwork.fromName(networkName);
+  }
+  return;
+};
+
+export const useRpcPayload = () => {
+  const { messageType, message, network, appName, domain } = useRpcSignStacksMessageParams();
+
+  if (isUtf8MessageType(messageType)) {
+    return {
+      messageType: 'utf8' as const,
+      message,
+      network: getNetwork(network),
+      appName,
+    } satisfies Utf8Payload;
+  }
+  if (isStructuredMessageType(messageType)) {
+    if (!domain) return null;
+
+    return {
+      messageType: 'structured' as const,
+      message: deserializeCV(Buffer.from(message, 'hex')),
+      domain: deserializeCV(Buffer.from(domain, 'hex')),
+      network: getNetwork(network),
+      appName,
+    } satisfies StructuredPayload;
+  }
+  return null;
+};
+
+export function useRpcSignStacksMessageParams() {
+  // TODO: do we need this? If yes, should we implement this in more places?
+  useRejectIfLedgerWallet('stx_signTransaction');
```

Comment:
We have this on stacks transaction signing so i thought we might want to have this one here too. @kyranjamie can you, please, take a look if this is how we want to handle it for RPC requests? Or should i make it work for ledger too?

Reference: https://github.com/leather-wallet/extension/blob/feat/add-rpc-stacks-message-signing/src/app/pages/rpc-sign-stacks-transaction/use-rpc-sign-stacks-transaction.ts#L15-L17

Reply:
Would be great if it can work with Ledger so we have parity with legacy requests . Happy to chat through how to do this in more detail if you'd like.